recorder.go

```diff
@@ -74,24 +96,35 @@ func getHostName(rawSpan basictracer.RawSpan) string {
 }
 
 func getServiceName(rawSpan basictracer.RawSpan) string {
-	s := getStringTag(rawSpan, string(ext.Component))
-	if s == "" {
-		s = getStringTag(rawSpan, string(ext.PeerService))
-		if s == "" {
-			return sensor.serviceName
+	// ServiceName can be determined from multiple sources and has
+	// the following priority (preferred first):
+	//   1. If added to the span via the OT component tag
+	//   2. If added to the span via the OT http.url tag
+	//   3. Specified in the tracer instantiation via Service option
```

Comment:
surprisingly OT has no service naming yet. and for this reason, I would like to get the sdk migration in so that we can make the adaption of OT conventions to instana semantics identical across languages afterwards. In the end, an instana service name is derived on the backend based on properties of a reported span and not by the tracer.